lib/ansible/module_utils/eos.py

```diff
@@ -180,6 +186,17 @@ def _load_params(self):
             if key in NET_COMMON_ARGS.keys():
                 if not params.get(key) and value is not None:
                     params[key] = value
+
+        env_overrides = dict(
+            username='ANSIBLE_NET_USERNAME',
+            password='ANSIBLE_NET_PASSWORD',
+            ssh_keyfile='ANSIBLE_NET_SSH_KEYFILE',
+            authorize='ANSIBLE_NET_AUTHORIZE',
+            auth_pass='ANSIBLE_NET_AUTH_PASS',
+        )
+        for key, env_var in env_overrides.items():
+            if env_var in os.environ:
+                params[key] = os.environ[env_var]
```

Comment:
I don't think we want to have environment variables overriding parameters.  What is okay is for environment variables to set defaults if the parameters are not given.

To do that you probably want the parameters to set a default value of None in the argument_spec.  Then in here you can check whether they are None.  If so, then try to set from the environment.  If it's still None and you have a sensible default (doesn't look like we would here except maybe for ssh_keyfile) then set it to that value here.

Also, could shorten this code slightly like this:
``` python
for key, env_var in env_overrides.items():
    params[key] = os.environ.get(env_var, None)
```